File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int[] sorted = new int[m+n];
        int k = 0,i=0,j=0;
        while(i<m && j<n){
            if(nums1[i]<=nums2[j]){
                sorted[k++] = nums1[i++];
            }else{
                sorted[k++] = nums2[j++];
            }
        }
        
        while(i<m){
            sorted[k++] = nums1[i++];
        }
        while(j<n){
            sorted[k++] = nums2[j++];
        }
        
        for(i=0;i<m+n;i++){
            nums1[i] = sorted[i];
        }
    }
}

File: 0088-merge-sorted-array/0088-merge-sorted-array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        i = m-1
        j = n-1
        l = m+n-1
        while l >=0 :
            if j>=0 and i>=0:
                if nums2[j] > nums1[i]:
                    nums1[l] = nums2[j]
                    j = j-1
                else:
                    nums1[l] = nums1[i]
                    i = i-1
            elif j>=0:
                nums1[l] = nums2[j]
                j = j-1
            l = l-1
        
        